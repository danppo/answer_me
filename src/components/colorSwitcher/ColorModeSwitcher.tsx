import * as React from "react";
import {
  useColorMode,
  useColorModeValue,
  IconButton,
  IconButtonProps,
} from "@chakra-ui/react";
import { FaMoon, FaSun } from "react-icons/fa";
import styles from "./colorSwitcher.module.scss";

type ColorModeSwitcherProps = Omit<IconButtonProps, "aria-label">;

const ColorModeSwitcher: React.FC<ColorModeSwitcherProps> = (props) => {
  const { toggleColorMode } = useColorMode();
  const text = useColorModeValue("dark", "light");
  const SwitchIcon = useColorModeValue(FaMoon, FaSun);

  return (
    <div className={styles.colorSwitcher}>
      <IconButton
        size="md"
        fontSize="lg"
        variant="ghost"
        color="current"
        marginLeft="2"
        onClick={toggleColorMode}
        icon={<SwitchIcon />}
        aria-label={`Switch to ${text} mode`}
        {...props}
      />
    </div>
  );
};

export default ColorModeSwitcher;
