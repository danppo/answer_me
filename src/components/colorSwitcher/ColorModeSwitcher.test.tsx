import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import { render } from '../../test-utils';
import ColorModeSwitcher from './ColorModeSwitcher';

// beforeEach(() => render(<Title />));

const titleValue = 'test title';

describe('renders components:', () => {
  test('self', () => {
    render(<ColorModeSwitcher />);
    const currentElement = screen.getByTestId('colorSwitcherDiv');
    expect(currentElement).toBeInTheDocument();
  });

  describe('test color switch action:', () => {
    render(<ColorModeSwitcher />);
    const button = screen.getByTestId('switcherButton');
    test('default dark ', () => {
      expect(button).toMatchSnapshot();
    });
    test('switch mode ', () => {
      fireEvent.click(button);
      expect(button).toMatchSnapshot();
    });
  });
});
