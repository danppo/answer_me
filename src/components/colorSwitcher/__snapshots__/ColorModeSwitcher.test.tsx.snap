// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components: test color switch action: default dark  1`] = `
<button
  aria-label="Switch to dark mode"
  class="chakra-button css-14uqn0c"
  data-testid="switcherButton"
  type="button"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    focusable="false"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M9.5 2c-1.82 0-3.53.5-5 1.35 2.99 1.73 5 4.95 5 8.65s-2.01 6.92-5 8.65c1.47.85 3.18 1.35 5 1.35 5.52 0 10-4.48 10-10S15.02 2 9.5 2z"
    />
  </svg>
</button>
`;

exports[`renders components: test color switch action: switch mode  1`] = `
<button
  aria-label="Switch to dark mode"
  class="chakra-button css-14uqn0c"
  data-testid="switcherButton"
  type="button"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    focusable="false"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M9.5 2c-1.82 0-3.53.5-5 1.35 2.99 1.73 5 4.95 5 8.65s-2.01 6.92-5 8.65c1.47.85 3.18 1.35 5 1.35 5.52 0 10-4.48 10-10S15.02 2 9.5 2z"
    />
  </svg>
</button>
`;
